File: Server/src/main/content/data/consumables/effects/PrayerEffect.java
package content.data.consumables.effects;

import core.game.consumable.ConsumableEffect;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.Skills;

public class PrayerEffect extends ConsumableEffect {
    double base, bonus;
    public PrayerEffect(double base, double bonus){
        this.base = base;
        this.bonus = bonus;
    }
    @Override
    public void activate(Player p) {
        int level = p.getSkills().getStaticLevel(Skills.PRAYER);
        double amt = base + (level * bonus);
        p.getSkills().incrementPrayerPoints(amt);
    }
}

File: Server/src/main/content/data/consumables/effects/PrayerEffect.kt
package content.data.consumables.effects

import core.api.*
import core.game.consumable.ConsumableEffect
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills
import org.rs09.consts.Items
import kotlin.math.ceil
import kotlin.math.floor

class PrayerEffect @JvmOverloads constructor(var base: Double, var bonus: Double, private val roundUp: Boolean = false) : ConsumableEffect() {
    override fun activate(player: Player?) {
        if(player == null) return
        val level = getStatLevel(player, Skills.PRAYER)
        var b = bonus
        if(inInventory(player, Items.HOLY_WRENCH_6714))
            b += 0.02 // Leaving this in for futureproofing. Current update is for properly rounding down.
        val rawAmount = base + (level * b)
        val amount = if (roundUp) ceil(rawAmount) else floor(rawAmount)
        modPrayerPoints(player, amount)
    }
}